src/main/java/me/jellysquid/mods/lithium/mixin/world/fast_noise/MixinSurfaceChunkGenerator.java

```diff
@@ -76,22 +73,22 @@ private double sampleNoise(int x, int y, int z, double horizontalScale, double v
             double upperNoise = 0.0;
 
             for (int octave = 0; octave < 16; octave++) {
-                double scaledVerticalScale = verticalScale * frequency;
-                upperNoise += sampleOctave(this.upperInterpolatedNoise.getOctave(octave), x, y, z, horizontalScale, verticalScale, scaledVerticalScale, frequency);
-                lowerNoise += sampleOctave(this.lowerInterpolatedNoise.getOctave(octave), x, y, z, horizontalScale, verticalScale, scaledVerticalScale, frequency);
+                upperNoise += sampleOctave(this.upperInterpolatedNoise.getOctave(octave), x, y, z, horizontalScale, verticalScale, frequency);
+                lowerNoise += sampleOctave(this.lowerInterpolatedNoise.getOctave(octave), x, y, z, horizontalScale, verticalScale, frequency);
 
                 frequency /= 2.0;
             }
 
-            // Vanilla behavior, return interpolated version
-            return MathHelper.clampedLerp(lowerNoise / 512.0, upperNoise / 512, clampedInterpolation);
+            // Vanilla behavior, return interpolated noise
+            return MathHelper.lerp(clampedInterpolation, lowerNoise / 512.0, upperNoise / 512);
         }
     }
 
-    private static double sampleOctave(PerlinNoiseSampler sampler, int x, int y, int z, double horizontalScale, double verticalScale, double scaledVerticalScale, double frequency) {
+    private static double sampleOctave(PerlinNoiseSampler sampler, int x, int y, int z, double horizontalScale, double verticalScale, double frequency) {
+        double scaledVerticalScale = verticalScale * frequency;
         return sampler.sample(
-                OctavePerlinNoiseSampler.maintainPrecision((double)x * horizontalScale * frequency),
-                OctavePerlinNoiseSampler.maintainPrecision((double)y * verticalScale * frequency),
-                OctavePerlinNoiseSampler.maintainPrecision((double)z * horizontalScale * frequency), scaledVerticalScale, (double)y * scaledVerticalScale) / frequency;
+                OctavePerlinNoiseSampler.maintainPrecision(x * horizontalScale * frequency),
+                OctavePerlinNoiseSampler.maintainPrecision(scaledVerticalScale * frequency),
```

Comment:
This should be `y * scaledVerticalScale` 😛

Reply:
haha oops